ocaml/xapi/livepatch.ml

```diff
@@ -243,7 +252,20 @@ module XenLivePatch = struct
       ["upload"; name; livepatch_file]
     |> ignore ;
     Helpers.call_script !Xapi_globs.xen_livepatch_cmd ["replace"; name]
-    |> ignore
+    |> ignore ;
+    (* Unload obsolete livepatches.
+     * The status of these livepatches is changed to 'CHECKED' from 'APPLIED' when
+     * the latest livepatch is being applied.
+     *)
+    get_checked_livepatches ()
+    |> List.iter (fun (base_version, base_release, to_version, to_release) ->
+           let name =
+             Printf.sprintf "lp_%s-%s-%s-%s" base_version base_release
+               to_version to_release
+           in
+           Helpers.call_script !Xapi_globs.xen_livepatch_cmd ["unload"; name]
```

Comment:
Does the order in which these patches are unloaded matter, i.e. do you need to unload them in the reverse order in which they got loaded, or it doesn't matter.
@andyhhp @rosslagerwall ?

Reply:
It worked well from my test in which 3 simple livepatches were used. At least no errors were reported by the `xen-livepatch` tool.
But, yeah, it would be much better to confirm this from rationale.